libraries/Microsoft.Bot.Builder/Streaming/StreamingRequestHandler.cs

```diff
@@ -381,11 +383,24 @@ private async Task ReconnectAsync(IDictionary<string, string> requestHeaders = n
                 }
             }
 
+            // Set the authentication header if it wasn't passed in.
+            if (requestHeaders != null && !requestHeaders.ContainsKey("authentication") && !requestHeaders.ContainsKey("Authentication"))
```

Comment:
Declare constants. Also, can you use linq? i.e. requestHeaders.Any(h => ...)?

Reply:
Cant recall if headers is ienumerable or a closed collection

Reply:
I usually get my hand slapped for using linq in the SDK for making code harder to read, so I took it as a general rule. Happy to use it here!